Remotion/Web/Development.WebTesting/ControlObject.cs

```diff
@@ -46,10 +46,17 @@ protected ControlObject ([NotNull] ControlObjectContext context)
     /// <summary>
     /// Returns the control's HTML ID.
     /// </summary>
-    /// <exception cref="MissingHtmlException">If the <see cref="Scope"/>'s root DOM element does not have an ID attribute.</exception>
+    /// <exception cref="WebTestException">If the <see cref="Scope"/>'s root DOM element does not have an ID attribute.</exception>
     public string GetHtmlID ()
     {
-      return Scope.Id;
+      try
+      {
+        return Scope.Id;
+      }
+      catch (MissingHtmlException)
+      {
+        throw new MissingHtmlException ("The control object does not have an ID attribute.");

```

Comment:
locator

Reply:
Reset to the following:
```
public string GetHtmlID ()
{
  return Scope.Id;
}
```
Returns `string.Empty` if the Id is missing or empty. This could throw a `MissingHtmlException` if the scope could not be found. This implies that the control object could not be found. `ControlObjectContext.CloneForControl(...)` synchronizes `Context.Scope` upon context creation using `EnsureExistence()` (re-throwing `MissingHtmlException` or `AmbiguousException` as `WebTestException`).

Reply:
But doesn't MissingHtmlException.Message have useful information?

Reply:
Agreed to move this to a separate issue: An empty or missing ID should throw a `MissingHtmlException` in the context of a `ControlObject`.